internal/hardware/hardware.go

```diff
@@ -36,6 +70,26 @@ func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
 		}
 		hardwareInfo.Interfaces = append(hardwareInfo.Interfaces, newInterface)
 	}
+}
 
-	return &hardwareInfo, nil
+func (s *HardwareInfo) init() {
+	if s.Dependencies == nil {
+		s.Dependencies = util.NewDependencies("/")
+	}
+}
+
+func (s *HardwareInfo) GetMutableHardwareInfoDelta(hardwareMutableInfoSource models.HardwareInfo, hardwareMutableInfoTarget models.HardwareInfo) *models.HardwareInfo {
```

Comment:
in terms of naming, the relation between the parameters isn't a source and a target.
it is prev/new or former/updated.

Reply:
Ok, if you feel that it would be more understandable, I'll do the change
FYI, I named target and source as it's checking the source variable against the target variable and only the values of the target that are different will be kept.

Reply:
I don't have a strong opinion about it. it can remain as is.